In [1]:
#time module import
import time
#numpy import
import numpy as np

In [2]:
#program for multiplying two matrices
def matrixmultiply(A, B):
    if len(A[0]) != len(B):
        return "Error"
    result = []
    for i in range(len(A)):
        row = []
        for j in range(len(B[0])):
            sum = 0
            for k in range(len(B)):
                sum += A[i][k] * B[k][j]
            row.append(sum)
        result.append(row)
    return result

In [3]:
#program for multiplying two matrices using numpy
def multiply_matrices(A, B):
   
    return np.dot(A, B)

In [4]:
A = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
B = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
#starting to record time
start = time.time()
print(matrixmultiply(A, B))
end = time.time()

print("The time of execution of above program is :",
      (end-start) * 10**3, "ms")

[[30, 36, 42], [66, 81, 96], [102, 126, 150]]
The time of execution of above program is : 0.28896331787109375 ms


In [6]:
A = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
B = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
#starting to record time
start = time.time()
print(multiply_matrices(A, B))
end = time.time()

print("The time of execution of above program is :",
      (end-start) * 10**3, "ms")

[[ 30  36  42]
 [ 66  81  96]
 [102 126 150]]
The time of execution of above program is : 0.19478797912597656 ms


Matrix multiplication using lists of list took approx. 0.28896 ms where as using numpy it took 0.19478 ms.

Numpy is faster that is because numpy integrates packages C, C++ codes in python which have a lesser execution time than python.